src/aes.rs

```diff
@@ -207,23 +212,25 @@ fn aes128_counter_mode(key: Key, nonce: Nonce, counter: u32, msg: ByteSlice) ->
     for i in 0..n_blocks {
         let keyblock = aes128_ctr_keyblock(key, nonce, ctr);
         let k = i * BLOCKSIZE;
-        blocks_out.update_vec(k, xor_block(msg.get(k..k + BLOCKSIZE), keyblock).into());
-        ctr += 1;
+        blocks_out = blocks_out.update(
+            k,
+            xor_block(Block::from_sub_pad(msg.clone(), k..k + BLOCKSIZE), keyblock),
```

Comment:
So... I'd really like to avoid having to call `clone`. The `ByteSlice` is a relatively ugly workaround for that and I'm open to other suggestions. But I'd like to avoid `clone` calls (unless you all vote for allowing them).

Reply:
I vote for allowing them. Actually, they are stricly needed. Indeed, Rust has a part of linearity in its type system. By forbidding borrowing, we force all functions to "consume" their arguments by moving them. We get around that with fixed-size arrays because they implement the `Copy` trait, but  variable-lenght arrays will never implement the `Copy` trait. So we have to clone them whenever using them. I expect variable-lenght arrays to be limited to input/output data and follow a relatively linear flow inside the program, so the number of `clone()` calls should be limited.

Reply:
Right I don't think we can get rid of cloning in general but I still think it would be nice to have some syntactical sugar to hide the `clone`. Lets merge like this and defer the decision on how to do it.